src/lib/modernExtend.ts

```diff
@@ -1243,12 +1243,19 @@ export function commandsColorCtrl(args?: CommandsColorCtrl): ModernExtend {
 
 export interface LockArgs {
     pinCodeCount: number;
+    endpointNames?: string[];
```

Comment:
I don't understand how this works, this is an array and `endpointName` is a `string` in `publish.ts`. Did you check if this works?

Reply:
Thanks much for your review 👍 

Actually, I tested it succesfully with the diy device describe in [issue #24352](https://github.com/Koenkk/zigbee2mqtt/issues/24352), that is to say made of the following 3 endpoints:
- Endpoint 8 : genBasic, genIdentify, msTemperatureMeasurement.
- Endpoint 10 : closuresWindowCovering ( `ep1` ) --> It uses key `state`
- Endpoint 12 : genOnOff ( `relay` ) --> It uses also key `state`

Your remark is more than relevant : My bad ! It works only if `endpointNames: string[]` contains only 1 item for sure :confused:

I believe `endpointNames` must definitely remain an array to allow multiple endpoints with the same cluster. Then I am gonna see how I can handle that in `publish.ts`. Please advise if you see a better way.

Reply:
We should update `publish.ts` to check if the endpoint in the array is endpoint is a list.

Reply:
:ok_hand: I will come back to you with a PR to update `publish.ts`

Reply:
[PR #24525](https://github.com/Koenkk/zigbee2mqtt/pull/24525) created.